File: app/src/main/java/com/example/pdyf/MainActivity.java
package com.example.pdyf;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.example.pdyf.DateBase.DataBaseHandlerCategory;
import com.example.pdyf.Goals.ActivityGoal;
import com.example.pdyf.Report.ActivityChart;
import com.example.pdyf.TransactionManager.Calculator.ActivityCalculator;
import com.example.pdyf.TransactionManager.Categories.Category;
import com.example.pdyf.TransactionManager.Transactions.ActivityAddTransaction;

import java.util.List;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        DataBaseHandlerCategory dataBaseHandlerCategory = new DataBaseHandlerCategory(this);
        dataBaseHandlerCategory.getWritableDatabase();
        List<Category> categoryList = dataBaseHandlerCategory.getAllCategories();

        // вывод бд в консоль
        for (Category category : categoryList) {
            Log.d("Categories info: ", "Id " + category.getId() + ", Name: " + category.getName() + ", Amount: " + category.getTotalSpent());
        }

    }

    public void onClick(View view) {
        Intent intent = new Intent(this, ActivityChart.class);
        startActivity(intent);
    }
    public void goal(View v){
        Intent intent = new Intent(this, ActivityGoal.class);
        startActivity(intent);
    }
    public void addTransaction(View v){
        Intent intent = new Intent(this, ActivityAddTransaction.class);
        startActivity(intent);
    }
    public void calculate(View v){
        Intent intent = new Intent(this, ActivityCalculator.class);
        startActivity(intent);
    }

    public void search(View v){
        Intent intent = new Intent(this, ActivitySearch.class);
        startActivity(intent);
    }



}

File: app/src/main/java/com/example/pdyf/MainActivity.kt
//package com.example.piechartdemo
//
//import android.os.Bundle
//import androidx.appcompat.app.AppCompatActivity
//import com.anychart.AnyChart
//import com.anychart.AnyChartView
//import com.anychart.chart.common.dataentry.DataEntry
//import com.anychart.chart.common.dataentry.ValueDataEntry
//import com.anychart.charts.Pie
//
//import com.example.pdyf.R
//
//class MainActivity : AppCompatActivity() {
//    private lateinit var anyChartView: AnyChartView
//
//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)
//
//        anyChartView = findViewById(R.id.any_chart_view)
//
//        // Настройка круговой диаграммы
//        setupPieChart()
//    }
//
//    private fun setupPieChart() {
//        // Создание объекта диаграммы
//        val pie: Pie = AnyChart.pie()
//
//        // Создание списка данных для диаграммы
//        val data = arrayListOf<ValueDataEntry>(
//            ValueDataEntry("First", 200),
//            ValueDataEntry("Second", 300),
//            ValueDataEntry("Third", 400)
//        )
//
//        // Добавление данных в диаграмму
//        pie.data(data as List<DataEntry>?)
//
//        // Установка диаграммы в AnyChartView
//        anyChartView.setChart(pie)
//
//        // Дополнительные настройки (например, заголовок)
//        pie.title("Пример круговой диаграммы")
//    }
//}